### ISSUE TITLE
Add support for Imaginary

### ISSUE THREAD

**Issue by enoch85** [ariselseng/camerarawpreviews#92]
Nextcloud recently move towards Imaginary, and it would be great if this app could support that instead of Imagick.

https://github.com/nextcloud/server/issues/13099
https://okxo.de/speed-up-nextcloud-preview-generation-with-imaginary/

**Comment by ariselseng** [comment]
Imagick is optional and only used for embedded TIFF previews.

**Comment by enoch85** [comment]
Still, think that Imagick should be replaced with Imaginary since that's where Nextcloud is going in general. :)

**Comment by ariselseng** [comment]
@enoch85 With Imaginary, the previews is not cached in Nextcloud appdata. Is that correct?
I see that the default server/lib/private/Preview/Imaginary.php does not support tiff.

It would be awesome if Imaginary could support raw files (maybe it can via extensions ?). Then this app could even fallback to Imaginary for raw files without an embedded jpeg file. There is always files that does have a preview, and a fallback to a http service to actually convert the raw image to jpg would be so nice. Then this app would be complete and actually work for all raw files. 
An idea I just got would be to setup a docker container, similar to imaginary, but with a simple http service that would call cli commands (or using some library) to convert the raw image to jpg and serve it.

**Comment by enoch85** [comment]
> With Imaginary, the previews is not cached in Nextcloud appdata. Is that correct?

It is. All the previews that are generated are done so on the fly, but then saved in cache for later use.

> An idea I just got would be to setup a docker container, similar to imaginary, but with a simple http service that would call cli commands (or using some library) to convert the raw image to jpg and serve it.

That sounds great!

Maybe even better would be to talk to the core team devs of Nextcloud and make it happen in the actual core? Also cc @szaimen on this.

**Comment by enoch85** [comment]
> > Maybe even better would be to talk to the core team devs of Nextcloud and make it happen in the actual core? Also cc @szaimen on this.
> 
> You mean having a provider that can connect to this http service, just like done for Imaginary in core?

Yeah something like that. Or would it possible to contribute to Imaginary to support RAW? Better to have one service than several, and since Imaginary now is the new "standard", maybe that would be the way forward?

**Comment by dewi-ny-je** [comment]
I found this ticket while looking for imagick in Nextcloud Snap, which does not provide it by default and therefore makes it impossible for me to use the related feature.
The reason why imagick is not provided anymore is that it has a terrible history of security, so replacing it completely or at least stopping using it should be considered:
https://github.com/nextcloud-snap/nextcloud-snap/issues/882

**Comment by ariselseng** [comment]
@dewi-ny-je Most of this apps functionality works with gd and imagick is not a requirement.

**Comment by Forza-tng** [comment]
AFAIK, Imaginary does not support any raw formats such as DNG, CR2 or NEF. However, would it be possible to use this app as a filter that sends a tiff/PNG file to Imaginary for image preview generation?

**Comment by JanisPlayer** [comment]
Just wanted to say there’s little hope, the project has been discontinued:
[https://github.com/nextcloud/all-in-one/discussions/1908](https://github.com/nextcloud/all-in-one/discussions/1908)

[https://github.com/h2non/imaginary](https://github.com/h2non/imaginary)

> imaginary can read images from HTTP POST payloads, server local paths or remote HTTP servers, supporting JPEG, PNG, WEBP, HEIF, and optionally TIFF, PDF, GIF and SVG formats if libvips@8.3+ is compiled with proper library bindings.

RAW isn’t supported either.
I also discussed changes to the project back then regarding AVIF, but maintaining it has become too time-consuming, so no further changes will be made.
